pkg/driver/driver_gcp.go

```diff
@@ -84,16 +84,29 @@ func (d *GCPDriver) Create() (string, string, error) {
 
 	var disks = []*compute.AttachedDisk{}
 	for _, disk := range d.GCPMachineClass.Spec.Disks {
-		disks = append(disks, &compute.AttachedDisk{
-			AutoDelete: disk.AutoDelete,
-			Boot:       disk.Boot,
-			InitializeParams: &compute.AttachedDiskInitializeParams{
-				DiskSizeGb:  disk.SizeGb,
-				DiskType:    fmt.Sprintf("zones/%s/diskTypes/%s", zone, disk.Type),
-				Labels:      disk.Labels,
-				SourceImage: disk.Image,
-			},
-		})
+		var attachedDisk compute.AttachedDisk
+		if disk.Type == "SCRATCH" {
+			attachedDisk = compute.AttachedDisk{
+				AutoDelete: disk.AutoDelete,
```

Comment:
This default to `false`, right?
@hardikdr @prashanth26 - Wouldn't it lead to similar issues that we recently saw of orphan disks when a VM is deleted.

Reply:
Good catch. What would be the best way to handle this? Default the values to true on MCM or Gardener maybe?

Reply:
IMO, defaulting should be done in MCM, keeping it mind that it can be used independently of Gardener

Reply:
@tennix - Can we default this flag to true in the MCM?

Reply:
I don't think it's related to this. Unlike to persistent disks, the local ssd disks will be deleted once the instance is deleted. But I can change the default value to `true` if you like.

Reply:
I'd actually prefer to delete these disks by default with the VM, unless specified in machinieclass. Considering the fact that we constantly create/delete the machines with autoscaler, we may end up with many such orphan disks, and we don't have active automation to keep track of these disks later.

Reply:
According to the documentation the local-ssds are expected to be deleted with the VM termination. - https://cloud.google.com/compute/docs/disks#localssds . 
- @tennix please correct if that's not the case.

Though, to save us from the possible changes at gcp's behaviour, I would prefer to set it to `true`. 
- The issue actually is, the API type for `AutoDelete` is defined as `bool` and leaves no good way to check if it's set at MachineClass or not. 
- @tennix Please feel free to change the type to `*bool` from `bool`, that'll help you in checking `AutoDelete==nil`(implies not set at MachineClass), and default to true.

Reply:
Okay, I'll change default `AutoDelete` to `true`.